### ISSUE TITLE
Row access policy always marked as changed when using heredoc style SQLs

### ISSUE THREAD

**Issue by fredriv** [Snowflake-Labs/terraform-provider-snowflake#2053]
**Provider Version**

0.70.1

**Terraform Version**

1.5.4

**Describe the bug**

When configuring the row access expression of a `snowflake_row_access_policy` using heredoc instead of a regular string, the resource is always marked as changed by `terraform plan`/`apply`.

**Expected behavior**

I expect the resource to not be marked as changed after applying it.

**Code samples and commands**

Example config to trigger the issue:
```
resource "snowflake_row_access_policy" "example_row_access_policy" {
  name     = "EXAMPLE_ROW_ACCESS_POLICY"
  database = "EXAMPLE_DB"
  schema   = "EXAMPLE_SCHEMA"
  signature = {
    A = "VARCHAR",
    B = "VARCHAR"
  }
  row_access_expression = <<-EOT
    case
      when current_role() in ('ANALYST') then true
      else false
    end
EOT
}
```

**Additional context**

I suspect this is a similar issue as in https://github.com/Snowflake-Labs/terraform-provider-snowflake/issues/534?

**Comment by fredriv** [comment]
Or perhaps related to https://github.com/Snowflake-Labs/terraform-provider-snowflake/issues/1422 / https://github.com/Snowflake-Labs/terraform-provider-snowflake/issues/1097?

**Comment by fredriv** [comment]
Verified that it's an issue with trailing newline in the heredoc style row access expression, similar to #1422 / #1097.

I moved the expression to a local variable and changed the row access policy resource to use row_access_expression = trim(local.expression, "\n") and the issue disappeared.

**Comment by sfc-gh-jmichalak** [comment]
Hi @fredriv 👋 
Sorry for a late response from our side. We are currently reworking row access policies. The issues you mentioned are related to other resources, but https://github.com/Snowflake-Labs/terraform-provider-snowflake/issues/1151 is related to row access policies as well. To be sure, could you paste `terraform apply` output?

**Comment by sfc-gh-jmichalak** [comment]
Hi, we've released [v0.96.0](https://github.com/Snowflake-Labs/terraform-provider-snowflake/releases/tag/v0.96.0), which includes adjusting and fixing row access policies. Please upgrade with the [migration guide](https://github.com/Snowflake-Labs/terraform-provider-snowflake/blob/main/MIGRATION_GUIDE.md#v0950--v0960).

**Comment by sfc-gh-jmichalak** [comment]
Closing the issue due to inactivity. Please create a new one if the issue persists in the newest version of the provider.
